packages/nextra-theme-docs/src/contexts/active-anchor.tsx

```diff
@@ -42,8 +41,12 @@ export const ActiveAnchorProvider = ({
   children: ReactNode
 }): ReactElement => {
   const [activeAnchor, setActiveAnchor] = useState<ActiveAnchor>({})
-  const observerRef = useRef<IntersectionObserver | null>(null)
-  if (IS_BROWSER && !observerRef.current) {
+  const observerRef = useRef<IntersectionObserver>(null!)
+
+  useEffect(() => {
+    const navbarHeight = getComputedStyle(document.body).getPropertyValue(
+      '--nextra-navbar-height'
+    )
```

Comment:
let's stay with navbar height value, because `nextraContentEl.offsetTop` can be 104px (navbar + banner) for this situation

<img width="1645" alt="image" src="https://github.com/user-attachments/assets/6718f7bf-b668-4830-8033-2a5ca388932a">

Reply:
Thank you for your suggestion. I hadn't considered this situation. The reason I used `offsetTop` is that the navbar could be configured by the user through `themeConfig`.

Reply:
oh yeah, in situation when user override navbar it could be not consistent with `--nextra-navbar-height` value